tox.ini

```diff
@@ -1,60 +1,30 @@
 [tox]
 envlist =
-    py27,py34,py35,py36,py37,pypy,
-    py27-pyramid{12,13,14,15,16,17,18,19},
+    py34,py35,py36,py37,py38,py39,pypy3,
     docs,
-    {py2,py3}-cover,coverage
+    coverage
 
 [testenv]
 # Most of these are defaults but if you specify any you can't fall back
 # to defaults for others.
 basepython =
-    py27: python2.7
     py34: python3.4
     py35: python3.5
     py36: python3.6
     py37: python3.7
     py38: python3.8
-    pypy: pypy
+    py39: python3.9
     pypy3: pypy3
-    py2: python2.7
-    py3: python3.5
-
-deps =
-    pyramid12: pyramid <= 1.2.99
-    pyramid13: pyramid <= 1.3.99
-    pyramid14: pyramid <= 1.4.99
-    pyramid15: pyramid <= 1.5.99
-    pyramid16: pyramid <= 1.6.99
-    pyramid17: pyramid <= 1.7.99
-    pyramid18: pyramid <= 1.8.99
-    pyramid19: pyramid <= 1.9.99
```

Comment:
To make it more clear, for @stevepiercy as well as others. These are the lines I am suggesting should NOT go. These allow us to test against a particular version of Pyramid.

Reply:
Find the version of Pyramid that first supported Py3 and drop the rest, sure, that's fine by me, but we shouldn't stop testing against Pyramid (and in all honesty, we should add pyramid 1.10 and 2.0 to this list)

Reply:
[Python 3 compatibility was added in Pyramid 1.3](https://docs.pylonsproject.org/projects/pyramid/en/latest/whatsnew-1.3.html).

Later versions of Pyramid dropped EOLed versions of Python 3, and we no longer maintain Pyramid 1.9 and older. I don't understand why we should test either versions of Pyramid that support only EOLed versions of Python or unsupported versions of Pyramid. Can you explain?

I agree with adding 1.10 and 2.0.

Reply:
I removed these lines as they were only used to for a matrix with the `py27` factor. Is this for ease of local testing?

Reply:
We use pyenv and tox, so local testing is not a problem.

One more data point for this discussion, the first version of Pyramid that supports Python 3.6 (the oldest Python not EOLed) was [Pyramid 1.8](https://docs.pylonsproject.org/projects/pyramid/en/latest/whatsnew-1.8.html#minor-feature-additions).

At the very least, we should have tests run on the most recent version of Python and supported versions of Pyramid:

```
py39-pyramid{110,20}
# ...
pyramid110: pyramid <= 1.10.99
pyramid20: pyramid <= 2.0.99
```

Reply:
I've added this to `tox.ini` and the Travis file. Previously no Python 3 testing was matrixed with the Pyramid version, so the pyramid versions above were removed with the Python 2.7 they were crossed against.

So cross testing with Pyramid was removed (as Python 2.7 was removed entirely), but no cross testing with Python 3.x + Pyramid was removed.

Reply:
These lines are necessary though so that during cross testing the right version of Pyramid is installed. By removing them entirely you have disabled the new newly added cross testing against the appropriate version of Pyramid.

py39-pyramid110 tells Tox it needs to install the dependencies as appropriate for the `pyramid110` matrix, and i don't see any lines added to add pyramid110/pyramid20 to the deps section.

Reply:
@stevepiercy already provided the appropriate lines that need to be added to the `deps` section: 


```
pyramid110: pyramid <= 1.10.99 
pyramid20: pyramid <= 2.0.99
```

Reply:
Uhhhh, Github is not showing your latest changes in the review tab for some reason :/ I see you already did this in a later commit.